modules/imap/hm-imap.php

```diff
@@ -829,6 +831,17 @@ public function get_message_list($uids, $raw=false) {
                                 }
                             }
                         }
+                        elseif ($vals[$i] == 'BODY[0.1') {
+                            $content = '';
+                            $i++;
+                            $i++;
```

Comment:
Seems strange but you probably tested this? Why do you increment so many times?

Reply:
That's right, it's just to allow us to access the message index while our main "for" loop is iterating.